### ISSUE TITLE
sql: casting timestamptz to timestamp not respecting timezone

### ISSUE THREAD

**Issue by eldos-dl** [cockroachdb/cockroach#16228]
**BUG REPORT**
Version:
```
Build Tag:    v1.0.1
Build Time:   2017/05/25 15:17:49
Distribution: CCL
Platform:     linux amd64
Go Version:   go1.8.1
C Compiler:   gcc 6.3.0
Build SHA-1:  50fd18dcf8da75a0f4980344a885fe6105d0cf23
Build Type:   release
```
Code:
```
SET TIME ZONE 'Asia/Kolkata';
CREATE TABLE a (
	id INT NOT NULL DEFAULT unique_rowid(), 
	t TIMESTAMP WITH TIME ZONE NOT NULL, 
	CONSTRAINT "primary" PRIMARY KEY (id ASC));
INSERT INTO a(t) VALUES (TIMESTAMPTZ '2017-01-01 00:00:00');
SELECT extract('year', t::TIMESTAMP) FROM a; -- Returns 2016
SELECT extract('year', t::DATE) FROM a; -- Returns 2017
```
Expected behaviour: Conversion of timestamptz to timestamp is expected to respect time zone as Postgres does that.

**Comment by petermattis** [comment]
@dt for triage, because I know you love TIMESTAMP issues.

**Comment by danhhz** [comment]
cc @justinj who apparently also knows about timestamps now

**Comment by dt** [comment]
@justinj i seem to recall we discussed something similar to this last month -- about some casts that should / should not take the session tz into account. do you remember what the conclusion was and if it is applicable here?

**Comment by justinj** [comment]
That was resolved in #18164, and was specifically related to parsing dates from strings. I checked out this behaviour and it seems like indeed this was not fixed by that change.
